With some Python versions: TypeError: TemporaryDirectory.__init__() got an unexpected keyword argument 'delete'

```
$ ./attotree \
            -t 8 \
            -L <(head -n 501  ngono.list) \
            -o ngono.n501.nw
        
[attotree] 2024-04-09 01:39:41 Attotree starting
Traceback (most recent call last):
  File "/Users/karel/github/my/attotree-paper/experiments/01_scaling_time/./attotree", line 461, in <module>
    main()
  File "/Users/karel/github/my/attotree-paper/experiments/01_scaling_time/./attotree", line 452, in main
    attotree(
  File "/Users/karel/github/my/attotree-paper/experiments/01_scaling_time/./attotree", line 298, in attotree
    with tempfile.TemporaryDirectory(delete=not debug) as d:
TypeError: TemporaryDirectory.__init__() got an unexpected keyword argument 'delete'
[Tue Apr  9 01:39:41 2024]
Error in rule attotree:
    jobid: 31
    input: ngono.list
    output: ngono.n501.nw
    shell:
        
        ./attotree \
            -t 8 \
            -L <(head -n 501  ngono.list) \
            -o ngono.n501.nw
        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job attotree since they might be corrupted:
ngono.n501.nw
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-04-09T013940.592027.snakemake.log
WorkflowError:
At least one job did not complete successfully.
make: *** [all] Error 1

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -4,7 +4,11 @@ on: [push]
 
 jobs:
   build-linux:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest]
+        python-version: [ 3.8, 3.9, "3.10", 3.11 ]
     strategy:
       max-parallel: 5
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -9,8 +9,6 @@ jobs:
       matrix:
         os: [ ubuntu-latest, macos-latest]
         python-version: [ 3.8, 3.9, "3.10", 3.11 ]
-    strategy:
-      max-parallel: 5
 
     steps:
     - uses: actions/checkout@v3

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -3,7 +3,7 @@ name: Python Package using Conda
 on: [push]
 
 jobs:
-  build-linux:
+  build:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
@@ -11,11 +11,11 @@ jobs:
         python-version: [ 3.7, 3.8, 3.9, "3.10", 3.11 ]
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python
-      uses: actions/setup-python@v3
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
       with:
-        python-version: {{ matrix.python }}
+        python-version: ${{ matrix.python-version }}
     - name: Add conda to system path
       run: |
         # $CONDA is an environment variable pointing to the root of the miniconda directory

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -8,14 +8,14 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest, macos-latest]
-        python-version: [ 3.8, 3.9, "3.10", 3.11 ]
+        python-version: [ 3.7, 3.8, 3.9, "3.10", 3.11 ]
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python 3.10
+    - name: Set up Python
       uses: actions/setup-python@v3
       with:
-        python-version: '3.10'
+        python-version: {{ matrix.python }}
     - name: Add conda to system path
       run: |
         # $CONDA is an environment variable pointing to the root of the miniconda directory

--- a/README.rst
+++ b/README.rst
@@ -1,6 +1,15 @@
 Attotree
 ========
 
+.. |github-release-badge| image:: https://img.shields.io/github/release/karel-brinda/attotree.svg
+    :target: https://github.com/karel-brinda/attotree/releases/
+.. |doi-badge| image:: https://zenodo.org/badge/DOI/110.5281/zenodo.10945896.svg
+    :target: https://doi.org/10.5281/zenodo.10945896
+.. |ci-tests-badge| image:: https://github.com/karel-brinda/attotree/actions/workflows/ci.yaml/badge.svg
+    :target: https://github.com/karel-brinda/attotree/actions/
+
+|info-badge| |paper-badge| |github-release-badge| |doi-badge| |ci-tests-badge|
+
 
 Introduction
 ------------

--- a/README.rst
+++ b/README.rst
@@ -1,14 +1,18 @@
 Attotree
 ========
 
+.. |info-badge| image:: https://img.shields.io/badge/Project-Info-blue
+    :target: https://github.com/karel-brinda/attotree
 .. |github-release-badge| image:: https://img.shields.io/github/release/karel-brinda/attotree.svg
     :target: https://github.com/karel-brinda/attotree/releases/
+.. |pypi-badge| image:: https://img.shields.io/pypi/v/attotree.svg
+    :target: https://pypi.org/project/attotree/
 .. |doi-badge| image:: https://zenodo.org/badge/DOI/110.5281/zenodo.10945896.svg
     :target: https://doi.org/10.5281/zenodo.10945896
 .. |ci-tests-badge| image:: https://github.com/karel-brinda/attotree/actions/workflows/ci.yml/badge.svg
     :target: https://github.com/karel-brinda/attotree/actions/
 
-|info-badge| |paper-badge| |github-release-badge| |doi-badge| |ci-tests-badge|
+|info-badge| |github-release-badge| |pypi-badge| |doi-badge| |ci-tests-badge|
 
 
 Introduction

--- a/README.rst
+++ b/README.rst
@@ -54,9 +54,9 @@ Command-line parameters
 .. code-block::
 
     $ attotree -h
- 
+
     Program: attotree (rapid estimation of phylogenetic trees using sketching)
-    Version: 0.1.3
+    Version: 0.1.4
     Author:  Karel Brinda <karel.brinda@inria.fr>
 
     usage: attotree [-k INT] [-s INT] [-t INT] [-o FILE] [-f STR] [-L] [-D] [-V] genomes [genomes ...]

--- a/README.rst
+++ b/README.rst
@@ -5,7 +5,7 @@ Attotree
     :target: https://github.com/karel-brinda/attotree/releases/
 .. |doi-badge| image:: https://zenodo.org/badge/DOI/110.5281/zenodo.10945896.svg
     :target: https://doi.org/10.5281/zenodo.10945896
-.. |ci-tests-badge| image:: https://github.com/karel-brinda/attotree/actions/workflows/ci.yaml/badge.svg
+.. |ci-tests-badge| image:: https://github.com/karel-brinda/attotree/actions/workflows/ci.yml/badge.svg
     :target: https://github.com/karel-brinda/attotree/actions/
 
 |info-badge| |paper-badge| |github-release-badge| |doi-badge| |ci-tests-badge|

--- a/attotree/attotree.py
+++ b/attotree/attotree.py
@@ -10,6 +10,7 @@ import argparse
 import datetime
 import os
 import re
+import shutil
 import subprocess
 import sys
 import tempfile
@@ -230,7 +231,7 @@ def quicktree(phylip_fn, newick_fn, algorithm, verbose):
     Returns:
         None
     """
-    message("Running quicktree")
+    message("Running Quicktree")
 
     cmd = "quicktree -in m".split()
     if algorithm == "upgma":
@@ -295,28 +296,32 @@ def attotree(fns, newick_fo, k, s, t, phylogeny_algorithm, fof, verbose, debug):
     else:
         fmsg = ""
     message(f"Attotree starting{fmsg}")
-    with tempfile.TemporaryDirectory(delete=not debug) as d:
-        message('Created a temporary directory', d)
-        phylip1_fn = os.path.join(d, "distances.phylip0")
-        phylip2_fn = os.path.join(d, "distances.phylip")
-        newick1_fn = os.path.join(d, "tree.nw")
-        newick2_fo = newick_fo
-        if fof:
-            #This is to make the list of file pass to Mash even with
-            #process substitutions
-            old_fof_fn = fns[0]
-            new_fof_fn = os.path.join(d, "fof.txt")
-            with open(old_fof_fn) as f, open(new_fof_fn, 'w') as g:
-                g.write(f.read())
-            fns = [new_fof_fn]
-        mash_triangle(fns, phylip1_fn, k=k, s=s, t=t, fof=fof, verbose=verbose)
-        postprocess_mash_phylip(phylip1_fn, phylip2_fn, verbose=verbose)
-        quicktree(phylip2_fn, newick1_fn, algorithm=phylogeny_algorithm, verbose=verbose)
-        postprocess_quicktree_nw(newick1_fn, newick2_fo, verbose=verbose)
+
+    d = tempfile.mkdtemp()
+
+    message('Creating a temporary directory', d)
+    phylip1_fn = os.path.join(d, "distances.phylip0")
+    phylip2_fn = os.path.join(d, "distances.phylip")
+    newick1_fn = os.path.join(d, "tree.nw")
+    newick2_fo = newick_fo
+    if fof:
+        #This is to make the list of file pass to Mash even with
+        #process substitutions
+        old_fof_fn = fns[0]
+        new_fof_fn = os.path.join(d, "fof.txt")
+        with open(old_fof_fn) as f, open(new_fof_fn, 'w') as g:
+            g.write(f.read())
+        fns = [new_fof_fn]
+    mash_triangle(fns, phylip1_fn, k=k, s=s, t=t, fof=fof, verbose=verbose)
+    postprocess_mash_phylip(phylip1_fn, phylip2_fn, verbose=verbose)
+    quicktree(phylip2_fn, newick1_fn, algorithm=phylogeny_algorithm, verbose=verbose)
+    postprocess_quicktree_nw(newick1_fn, newick2_fo, verbose=verbose)
 
     if debug:
         emsg = f" (auxiliary files retained in '{d}')"
     else:
+        shutil.rmtree(d)
+        message('Deleting the temporary directory', d)
         emsg = ""
     message(f"Attotree finished{emsg}")
 

--- a/attotree/version.py
+++ b/attotree/version.py
@@ -2,4 +2,4 @@ try:
     from __commit import *
 except ImportError:
     pass
-VERSION = "0.1.3"
+VERSION = "0.1.4"

--- a/tests/02_simple_tree/Makefile
+++ b/tests/02_simple_tree/Makefile
@@ -12,7 +12,12 @@ FASTAS=$(shell tar -tf test_spneumo.tar.xz)
 ## Compare trees ##
 ###################
 
-all: tree.precomputed.newick tree.default.nw tree.file_list.nw tree.file_list_proc_subst.nw
+all: \
+	tree.precomputed.newick \
+	tree.default.nw \
+	tree.default_verbose_debug.nw \
+	tree.file_list.nw \
+	tree.file_list_proc_subst.nw
 	for x in $^; do \
 		2>&1 echo Testing "$$x"; \
 		diff $$x tree.precomputed.newick; \
@@ -24,6 +29,10 @@ all: tree.precomputed.newick tree.default.nw tree.file_list.nw tree.file_list_pr
 
 tree.default.nw: file_list.txt
 	../../attotree/attotree.py $(shell cat $<) > "$@.tmp"
+	mv "$@.tmp" "$@"
+
+tree.default_verbose_debug.nw: file_list.txt
+	../../attotree/attotree.py -V -D $(shell cat $<) > "$@.tmp"
 	mv "$@.tmp" "$@"
 
 tree.file_list.nw: file_list.txt

